server/util/redact/redact_test.go

```diff
@@ -378,3 +378,51 @@ func TestRedactAPIKey(t *testing.T) {
 		require.Contains(t, json, notAPIKey)
 	}
 }
+
+func TestRedactRunResidual(t *testing.T) {
+	for _, tc := range []struct {
+		name     string
+		given    []string
+		expected []string
+	}{
+		{
+			name:     "WithoutAPIKey",
+			given:    []string{"-foo=bar", "-baz"},
+			expected: []string{"-foo=bar", "-baz"},
+		},
+		{
+			name:     "WithAPIKey_NoSplit",
+			given:    []string{"-api_key=foobar", "-baz"},
+			expected: []string{"-baz"},
+		},
+		{
+			name:     "WithAPIKey_Split",
+			given:    []string{"-api_key", "foobar", "-baz"},
+			expected: []string{"-baz"},
```

Comment:
I'm not sure if the result here could be confusing, should the expected result be

```
"-api_key", "<REDACTED>", "-baz"
```

and for the one above,

```
"-api_key=<REDACTED>", "-baz" 
```

Reply:
also might be reasonable to redact anything matching the regex `\b(key|token|secret)\b`

that way we redact `api-key`, `api_key`, `github_token`, `client_secret` etc.

Reply:
Hmm it might need to be `(\b|_)(key|token|secret)(\b|_)` instead because `\b` does not include underscore `_`? 🤔 

Let me look into it.

Reply:
Ah yeah. This should do the trick:

```
(^|[^a-z])(key|secret|token|password)($|[^a-z])
```

Reply:
I pushed a change to make use of the new regex, let me know what you think.